packages/media/src/utils/Safari17Handler.ts

```diff
@@ -0,0 +1,1031 @@
+
+            logger.debug("restartIce() | calling pc.setRemoteDescription() [offer:%o]", offer);
+
+            await this._pc.setRemoteDescription(offer);
+
+            const answer = await this._pc.createAnswer();
+
+            logger.debug("restartIce() | calling pc.setLocalDescription() [answer:%o]", answer);
+
+            await this._pc.setLocalDescription(answer);
+        }
+    }
+
+    async getTransportStats(): Promise<RTCStatsReport> {
+        this.assertNotClosed();
+
+        return this._pc.getStats();
+    }
+
+    async send({ track, encodings, codecOptions, codec, onRtpSender }: HandlerSendOptions): Promise<HandlerSendResult> {
+        this.assertNotClosed();
+        this.assertSendDirection();
+
+        logger.debug("send() [kind:%s, track.id:%s]", track.kind, track.id);
+
+        const sendingRtpParameters = utils.clone<RtpParameters>(this._sendingRtpParametersByKind![track.kind]);
+
+        // This may throw.
+        sendingRtpParameters.codecs = ortc.reduceCodecs(sendingRtpParameters.codecs, codec);
+
+        if (encodings && encodings.length > 1) {
+            // Set rid and verify scalabilityMode in each encoding.
+            // NOTE: Even if WebRTC allows different scalabilityMode (different number
+            // of temporal layers) per simulcast stream, we need that those are the
+            // same in all them, so let's pick up the highest value.
+            // NOTE: If scalabilityMode is not given, Chrome will use L1T3.
+            let maxTemporalLayers = 1;
+
+            for (const encoding of encodings) {
+                const temporalLayers = encoding.scalabilityMode
+                    ? parseScalabilityMode(encoding.scalabilityMode).temporalLayers
+                    : 3;
+
+                if (temporalLayers > maxTemporalLayers) {
+                    maxTemporalLayers = temporalLayers;
+                }
+            }
+
+            encodings.forEach((encoding, idx: number) => {
+                encoding.rid = `r${idx}`;
+                encoding.scalabilityMode = `L1T${maxTemporalLayers}`;
+            });
+        }
+
+        const sendingRemoteRtpParameters = utils.clone<RtpParameters>(
+            this._sendingRemoteRtpParametersByKind![track.kind],
+        );
+
+        // This may throw.
+        sendingRemoteRtpParameters.codecs = ortc.reduceCodecs(sendingRemoteRtpParameters.codecs, codec);
+
+        const mediaSectionIdx = this._remoteSdp!.getNextMediaSectionIdx();
+        let transceiver: RTCRtpTransceiver;
+        try {
+            transceiver = this._pc.addTransceiver(track, {
+                direction: "sendonly",
+                streams: [this._sendStream],
+                sendEncodings: encodings,
+            });
+        } catch (e) {
+            if ((e as Error).message === "Type error") {
+                console.warn("addTransceiver with encodings failed, retrying without encodings");
+                transceiver = this._pc.addTransceiver(track, {
```

Comment:
Maybe if we get here we should do something like `encodings = undefined`
As later we use the presence of that array to do some other stuff. 

But this is only here to fix a problem with the Playwright Webkit browser. using `sendEncodings` in `addTransceiver` works fine in 'real' safari 17.4 and 18.1.1, so maybe we don't need to worry about it?